src/Storages/RabbitMQ/WriteBufferToRabbitMQProducer.cpp

```diff
@@ -52,47 +53,38 @@ WriteBufferToRabbitMQProducer::WriteBufferToRabbitMQProducer(
 
     loop = std::make_unique<uv_loop_t>();
     uv_loop_init(loop.get());
-
     event_handler = std::make_unique<RabbitMQHandler>(loop.get(), log);
-    connection = std::make_unique<AMQP::TcpConnection>(event_handler.get(), AMQP::Address(parsed_address.first, parsed_address.second, AMQP::Login(login_password.first, login_password.second), "/"));
-
-    /* The reason behind making a separate connection for each concurrent producer is explained here:
-     * https://github.com/CopernicaMarketingSoftware/AMQP-CPP/issues/128#issuecomment-300780086 - publishing from
-     * different threads (as outputStreams are asynchronous) with the same connection leads to internal library errors.
-     */
-    size_t cnt_retries = 0;
-    while (!connection->ready() && ++cnt_retries != RETRIES_MAX)
-    {
-        event_handler->iterateLoop();
-        std::this_thread::sleep_for(std::chrono::milliseconds(CONNECT_SLEEP));
-    }
 
-    if (!connection->ready())
-    {
-        throw Exception("Cannot set up connection for producer", ErrorCodes::CANNOT_CONNECT_RABBITMQ);
-    }
+    if (setupConnection(false))
```

Comment:
But what happens in the bad case? Maybe we should throw an exception?

Reply:
now it will continue to attempt reconnect in `writingFunc` and will set up channel for producer the moment it managed to do so, but yes, should better throw if unable to connect while in buffer constructor